src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -210,6 +210,9 @@ internal string FullPath
                 {
                     _solutionFile = value;
                     _solutionFilter = null;
+
+                    // Store the directory of the file as the current directory may change while we are processes the file
```

Comment:
Sorry I just copied this from here: https://github.com/dotnet/msbuild/blob/master/src/Build/Construction/Solution/SolutionFile.cs#L477-L478

The problem with the unit tests hitting NullRefs was because the SolutionFileDirectory is only set during parse.  But our unit tests don't always parse an entire solution as some of them only parse single lines so the SolutionFileDirectory property doesn't get set.  So I replicated setting this property here and copied the comment along with the line.  Should I just delete the comment?

Reply:
¯\_(ツ)_/¯

I didn't find anything that validated the comment, but that doesn't mean there was nothing to find. It's probably just out-of-date, so yes?